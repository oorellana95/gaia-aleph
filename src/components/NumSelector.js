import React from 'react'

function NumSelector(props) {
    const [val, setValue] = React.useState(1)
    return (
        <>
            <div className="blog_right_sidebar">
                <aside className="single_sidebar_widget post_category_widget">
                    <h4 className="widget_title">Min. Passengers</h4>
                    <nav className="pagination pagination-info" aria-label="pagination">
                        <ul className="pagination-info pagination">
                            {props.nums.map(num => {
                                return (
                                    <li key={num} id={"li"+num} className = {'page-item ' + (num < val ? '' : 'active')} onClick={() => {setValue(num); props.filterFunction(num)}}>
                                        <span className="page-link">{num}</span>
                                    </li>
                                )
                            })}
                        </ul>
                    </nav>
                </aside>
            </div>
        </>
    );
}

export default NumSelector;