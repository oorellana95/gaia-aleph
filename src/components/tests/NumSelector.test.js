import React from 'react'
import NumSelector from '../NumSelector';
import { shallow } from 'enzyme';

describe('NumSelector from unit test', () => {
    test('it should render without crashing', () => {
        const wrapper = shallow(<NumSelector nums={[1,2,4,5,7]} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('it should crash if there is no array of nums or if it is not an array', () => {
        expect(() => {shallow(<NumSelector/>)}).toThrow();
        expect(() => {shallow(<NumSelector nums="Hola" />)}).toThrow();
        expect(() => {shallow(<NumSelector nums={4} />)}).toThrow();
    });
    

    test('should have length 5', () => {
        const wrapper = shallow(<NumSelector nums={[1,2,4,5,7]}/>);
        expect(wrapper.find('li')).toHaveLength(5)
    });

    test('should run onClick with the new state', () => {
        const updateFilter = jest.fn();
        const wrapper = shallow(<NumSelector nums={[1,2,4,5,7]} filterFunction={updateFilter}/>);
        //Si simulo un click en el li 1, la funcion updateFilter (filterFunction) se ha de ejecutar con el 1
        // target:{value: x}
        wrapper.find('#li1').simulate('click');
        expect(updateFilter).toHaveBeenCalledWith(1)
    });
});
