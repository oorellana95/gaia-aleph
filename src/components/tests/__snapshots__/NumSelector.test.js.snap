// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumSelector from unit test it should render without crashing 1`] = `ShallowWrapper {}`;

exports[`it should render without crashing 1`] = `ShallowWrapper {}`;
